### PR TITLE
Bug 1992596: e2e/cli: Move annotations.sh to proper e2e

### PR DESCRIPTION
**Pull request by deejross**
This moves the testing from a bash script to a Ginkgo test files.

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/approve
/assign @atiratree

File: test/extended/testdata/cmd/test/cmd/annotations.sh [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-# This test validates empty values in key-value pairs set by the annotate command
-os::cmd::expect_success_and_text 'oc create -f ${TEST_DATA}/hello-openshift/hello-pod.json' 'pod/hello-openshift created'
-os::cmd::expect_success_and_text 'oc annotate pod hello-openshift node-selector=""' 'pod/hello-openshift annotated'
-os::cmd::expect_success_and_not_text 'oc get pod hello-openshift --template="{{index .metadata.annotations \"node-selector\"}}"' '.'
-
-echo "annotate: ok"
-os::test::junit::declare_suite_end
-
-os::test::junit::declare_suite_start "cmd/label"
-# This test validates empty values in key-value pairs set by the label command
```
Comment (atiratree) [inline test/extended/testdata/cmd/test/cmd/annotations.sh:23]: can we also include the tests for empty values?

Comment (deejross) [reply test/extended/testdata/cmd/test/cmd/annotations.sh:23]: Latest commit includes removing annotations and labels and validating they are empty

Comment (atiratree) [reply test/extended/testdata/cmd/test/cmd/annotations.sh:23]: I was asking specifically for the test we have here (

File: test/extended/cli/annotation.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		out, err = oc.Run("annotate").Args("pod", "hello-openshift", "new-anno=hello").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.ContainSubstring("pod/hello-openshift annotated"))
+
+		g.By("validating new annotation")
+		out, err = oc.Run("get").Args("pod", "hello-openshift", "--template", podAnnotationTemplate).Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(out).To(o.Equal("hello"))
+
+		g.By("removing the annotation")
```
Comment (atiratree) [inline test/extended/cli/annotation.go:38]: +1, having additional test doesn't hurt anything

Comment (atiratree) [inline test/extended/cli/annotation.go:38]: +1, having additional test doesn't hurt anything

**Comment by yselkowitz** [comment]
The test as currently written points to the "openshift/hello-openshift" image, which is not multi-arch enabled.   This will cause these tests to fail in both disconnect installs, and on any install on non-x86 architectures.  Please rework this test according to the [procedures for e2e test images](https://github.com/openshift/origin/tree/master/test/extended/util/image).
/hold

**Comment by deejross** [comment]
@yselkowitz I ran into something similar with the next PR where the old bash scripts were using images that weren't allowed. That PR #26415 includes several fixes for these types of issues, and I'm considering closing this one and moving it over there to take advantage of the newer resource builder which uses approved e2e images.

**Comment by deejross** [comment]
This has been moved into #26415 to better handle multi-arch and disconnected installs.
